### ISSUE TITLE
Calling `channel.history` on Android causes an app crash

### ISSUE THREAD

**Issue by kj800x** [ably/ably-js-react-native#10]
Calling `channel.history` on an Ably Realtime's channel object causes a crash on Android.

Reproducable Example: https://github.com/kj800x/ReactNativeAbly2

There is no problem running this method on iOS.

When attempting to use the `channel.history` method on Android, the following stacktrace is generated:

```
getBoolean @ ReadableNativeArray.java:104
extractArgument @ JavaMethodWrapper.java:37
extractArgument @ JavaMethodWrapper.java:33
invoke @ JavaMethodWrapper.java:359
invoke @ JavaModuleWrapper.java:160
run @ NativeRunnable.java
handleCallback @ Handler.java:790
dispatchMessage @ Handler.java:99
dispatchMessage @ MessageQueueThreadHandler.java:29
loop @ Looper.java:164
run @ MessageQueueThreadImpl.java:192
run @ Thread.java:764
```

<img width="1680" alt="screen shot 2019-01-18 at 5 57 36 pm" src="https://user-images.githubusercontent.com/1657142/51417464-2264b300-1b4c-11e9-8db0-5287e815e789.png">

**Comment by paddybyers** [comment]
This looks similar to https://github.com/facebook/react-native/issues/20754, which is a null pointer exception, but this is a checkcast

**Comment by paddybyers** [comment]
The error is indicating that the javascript library is passing an argument to the React library passing a string for something that's expected to be a boolean.

I'm suspecting https://github.com/ably/ably-js/blob/master/browser/lib/transport/xhrrequest.js#L123

compare with: https://github.com/facebook/react-native/blob/master/Libraries/Network/XMLHttpRequest.js#L125

**Comment by paddybyers** [comment]
> I'm suspecting https://github.com/ably/ably-js/blob/master/browser/lib/transport/xhrrequest.js#L123

I can confirm that updating this to `xhr.withCredentials = true;` fixes the problem.

**Comment by SimonWoolf** [comment]
Fixed by https://github.com/ably/ably-js/commit/7290f5160b1bf294e6a42cc472c26a9cf494c950 , released in ably-js 1.0.23, ably-js-react-native version bumped to 1.0.1 to force ably-js >=1.0.23
